### PR TITLE
Implement beacon status icon/button

### LINKED ISSUE #1552: Request: Beacon Status Button in the Wallet
From cleaning the gridcoin-tasks issues.

> ###Spec:
> make a status monitor which functions also as a button that fits in the notification strip right next to locked, staking, up to date, etc. are.
> ###Suggestion:
> The button be shapped like a chainlink. Examples: image or image
> for examples.
> Let that status button be for joining (sending a beacon) and monitoring your beacon on the network.
> 
> ### Reason:
> Right now sending a beacon is cryptic for new users.
> Why not color code it clearly:
> 
> * Green - Beacon is on the network.
> * Yellow - Beacon is 5 months old and needs renewal.  (Click to resend your beacon, you will be prompted to unlock your wallet if necessary.)
> * Red - you have no beacon matching your current CPID on the network. (Click to send your beacon, you will be prompted to unlock your wallet if necessary.
> 
> **Bonus**: hover over the JOIN button and see your receiving address attached to your beacon.

https://github.com/gridcoin-community/Gridcoin-Tasks/issues/214

### PR DESCRIPTION
**Pull request by jamescowens**
This PR implements the beacon status "button" and closes #1552 when complete. I decided to use a lighthouse with beams, because that really makes you think about beacons, IMO.

Note that I have replaced all of the png icons for the vertical "status strip" with svg ones, and implemented the new scraper icons that toggleton and nethlek did a while back.

The svg icons scale much better on HiDPI displays than the original png ones.

The plan for the beacon icon is to have it have it
1. lighthouse with no beams for investor only, when that is intended
2. present with green beams when the beacon is valid and not in the renewal zone (i.e. < 150 days since last advertised/renewed).
3. present with yellow beams when the beacon is valid and due for renewal (i.e. >= 150 days but <= 165 days)
4. present with red beams when within 15 days of expiring (> 165 days) or not present when it should be (because email= and boinc indicate the CPID should be linked)

I will do a tooltip which provides an appropriate descriptive status.

If I have time, I will also implement a small wizard that will open if the lighthouse is clicked and the beacon needs to be advertised, is renewable, or is expired. The wizard will first prompt to unlock the wallet fully, do the advertisebeacon and then relock the wallet and prompt when the wallet should be unlocked for staking only.

Here is a screenshot of this initial stage of the PR.
image

### CODE REVIEW

**Comment by jamescowens** [comment]
@jring-o I think you will be pleased with this. Any additional suggestions welcome.

**Comment by cyrossignol** [comment]
I like the icon. I built a dialog for the new beacon workflow that behaves like the wizard you described. I can hook this up to the status icon.

**Comment by jamescowens** [comment]
Ok. Here is a screenshot for the tooltip for an active reseacher...
image

**Comment by jamescowens** [comment]
And an investor...
image

**Comment by jamescowens** [comment]
@cyrossignol , then I will stop here and go ahead and mark this ready for review. We can hook your wizard/dialog in when you are ready in another PR.

**Comment by jamescowens** [comment]
I refined some of the subcases... This is what the tooltip looks like if the wallet is in investor mode, but the email entry in the config exists...

image

File: src/qt/res/icons/x.svg [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (cyrossignol) [inline src/qt/res/icons/x.svg]: Perhaps we should give the file the same name as the alias. Took me a second to figure out what this is used for.

File: src/qt/bitcoingui.cpp [hunk 2]
```diff
@@ -1606,8 +1621,99 @@ void BitcoinGUI::updateScraperIcon(int scraperEventtype, int status)
     else if ((scraperEventtype == (int)scrapereventtypes::Convergence  || scraperEventtype == (int)scrapereventtypes::SBContract)
              && status == CT_DELETED)
     {
-        labelScraperIcon->setPixmap(QIcon(":/icons/quit").pixmap(STATUSBAR_ICONSIZE, STATUSBAR_ICONSIZE));
+        labelScraperIcon->setPixmap(QIcon(":/icons/red_and_white_x").pixmap(STATUSBAR_ICONSIZE, STATUSBAR_ICONSIZE));
```
[the remark this answers is not in the document]
Comment (jamescowens) [reply src/qt/bitcoingui.cpp]: Yep. Fixed.

File: src/qt/res/icons/x.svg (hunk 1 above)
Comment (jamescowens) [reply src/qt/res/icons/x.svg]: Done.

File: src/neuralnet/researcher.cpp [hunk 3]
```diff
@@ -507,6 +482,31 @@ std::string Researcher::Email()
     return email;
 }
 
+//!
+//! \brief Determine whether the wallet must run in investor mode before trying
+//! to load BOINC CPIDs.
+//!
+//! \return \c true if the user explicitly configured investor mode or failed
+//! to input a valid email address.
+//!
```
[the remark this answers is not in the document]
Comment (jamescowens) [reply src/neuralnet/researcher.cpp]: Done

File: src/neuralnet/researcher.cpp [hunk 4]
```diff
@@ -507,6 +482,31 @@ std::string Researcher::Email()
...
+bool Researcher::ConfiguredForInvestorMode()
+{
+    if (GetBoolArg("-investor", false)) {
+        LogPrintf("Investor mode configured. Skipping CPID import.");
```
[the remark this answers is not in the document]
Comment (jamescowens) [reply src/neuralnet/researcher.cpp]: That is the simplest approach. Done.

**Review by cyrossignol — APPROVED** [review]
Looks great!

**Comment by jamescowens** [comment]
This is a good one I think! :)

**Comment by 9swampy** [comment]
Nice. With the refactor to separate would it be easy to expose via an RPC call; or is it derivable already anyhow?

**Comment by jamescowens** [comment]
@9swampy what exactly do you need and I can do a PR for it.

**Comment by 9swampy** [comment]
You know i've got an RPC client that i'm polling to check the wallet is staking; would be good to be able to inform the beacon's up too. I didn't see any way to verify beacon status from available RPC responses; educate me if there is a way, and perhaps the wallet staking status would already reflect the beacon being down but it would be helpful to be able to forewarn impending beacon expiry.

If it wouldn't be too hard to expose the status you've exposed to the GUI through the RPC interface that's all I was suggesting.

**Comment by jamescowens** [comment]
It is easy and I will do it. The existing beaconstatus rpc call, which actually now calls the internal function GetBeaconStatus, is too heavyweight, so I will create another lighter one, or else put a parameter in there to thin it out... stay tuned.

**Pull request merged**